utils/sql-db-seeder/src/utilisation-report/utilisation-report.seed.ts

```diff
@@ -27,6 +27,9 @@ export default class UtilisationReportSeeder implements Seeder {
     const uploadedReportReportPeriod = getCurrentReportPeriodForBankSchedule(paymentReportOfficerBank.utilisationReportPeriodSchedule);
     const uploadedReport = createUploadedReport(paymentReportOfficer, uploadedReportReportPeriod, 'PENDING_RECONCILIATION');
 
+    // The reports need to be seeded either before or with the fee records
+    appendFeeRecordsToUtilisationReport(uploadedReport);
```

Comment:
Not a comment for a change, just a general comment re side effects :) was there a reason we did it this way (with side effects)?

I tend to dislike functions with side effects, even if we aren't strictly functional programming, as my default assumption is we don't have side effects to functions. I had to do a check here to make sure that we were modifying this object -- although that's probably just me being used to patterns without side effects ie:
`const uploadedReportWithFeeRecords = {...uploadedReport, feeRecords: createFeeRecordsForUtilisationReport(uploadedReport)}`
(or even, just combining this inside `createUploadReport`)

Reply:
Because this is a class, we can't just do `{ ...uploadedReport, feeRecords: foo }` as it will lose the methods that exist on the prototype. The way to do it without side effects would be to do
```typescript
const uploadedReportFeeRecords = createFeeRecords({ ... });
uploadedReport.feeRecords = uploadedReportFeeRecords;
```
which I am happy to do